src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -852,6 +854,92 @@ JNI_METHOD(void, readPath)
     callback->mReadClient = readClient;
 }
 
+JNI_METHOD(void, subscribeToEventPath)
+(JNIEnv * env, jobject self, jlong handle, jlong callbackHandle, jlong devicePtr, jobject eventPathList, jint minInterval,
+ jint maxInterval, jboolean keepSubscriptions, jboolean isFabricFiltered)
+{
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    DeviceProxy * device = reinterpret_cast<DeviceProxy *>(devicePtr);
+    if (device == nullptr)
+    {
+        ChipLogError(Controller, "No device found");
+        JniReferences::GetInstance().ThrowError(env, sChipDeviceControllerExceptionCls, CHIP_ERROR_INCORRECT_STATE);
+    }
+
+    std::vector<app::EventPathParams> eventPathParamsList;
+    err = ParseEventPathList(eventPathList, eventPathParamsList);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Error parsing Java event paths: %s", ErrorStr(err)));
+
+    app::ReadPrepareParams params(device->GetSecureSession().Value());
+    params.mMinIntervalFloorSeconds   = minInterval;
+    params.mMaxIntervalCeilingSeconds = maxInterval;
+    params.mpEventPathParamsList      = eventPathParamsList.data();
+    params.mEventPathParamsListSize   = eventPathParamsList.size();
+    params.mKeepSubscriptions         = static_cast<bool>(keepSubscriptions != JNI_FALSE);
```

Comment:
are the static_casts required here? `operator !=` seems like it would return a bool.

Reply:
I remove "static_cast".